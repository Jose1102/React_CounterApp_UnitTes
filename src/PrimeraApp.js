import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
// FC 
const PrimeraApp = ( {saludo , subtitulo } ) => {

    //const saludo = 'Esta es la constante del saludo';

    //console.log(props);

   

    return (
        <Fragment>
            <h1>{ "Hola soy "+saludo }</h1>
            <p>{ subtitulo }</p>

        </Fragment>
        
        
    );



}


PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired 

}

PrimeraApp.defaultProps = {
    subtitulo : 'Este es un subtitulo'
}



export default PrimeraApp;