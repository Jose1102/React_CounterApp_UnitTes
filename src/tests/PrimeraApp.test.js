import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom'


describe('prueba en <PrimeraApp />', ()=>{
    // test('debe mostrar el mensaje "Hola soy Goku"', () => {
    //     const saludo = "Hola soy Goku";
    //     const {getByText} = render( <PrimeraApp saludo="Goku"/> )
    //     expect( getByText( saludo )).toBeInTheDocument();        
    // })
    
    test('debe mostrar <PrimeraApp /> correctamente ', () => {
        const saludo = 'Goku'
        const wraper = shallow( <PrimeraApp saludo={saludo}/>);
        expect( wraper ).toMatchSnapshot();

    })

    test(' debe de mostrar el subtitulo mostrado por props ', () => {

        const saludo = 'Goku';
        const subTitulo = 'Soy un subtitulo';
        const wraper = shallow( 
            <PrimeraApp 
                saludo={saludo}
                subtitulo={ subTitulo }                
            />
        );

        const textoParrafo = wraper.find( 'p' ).text();
        console.log(textoParrafo);

        expect( textoParrafo ).toBe( subTitulo);
        
    })
    
    
})