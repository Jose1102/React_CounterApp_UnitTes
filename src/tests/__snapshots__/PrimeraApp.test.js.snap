// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola soy Goku
  </h1>
  <p>
    Este es un subtitulo
  </p>
</Fragment>
`;
